### ISSUE TITLE
Commit hashes of tagged versions? (Migration from Bitnami native install to docker)

### ISSUE THREAD

**Issue by fdcastel** [bitnami/bitnami-docker-phabricator#13]
Where I can get the exact git commit versions (for phabricator, arcanist and phutil) used to build each [tagged version](https://hub.docker.com/r/bitnami/phabricator/tags/) of phabricator docker images?

### Why this information is important?

I have a Phabricator install built using Bitnami native installer. This box is running fine in production for some months now. 

It was upgraded sometimes following the [official Bitnami instructions](https://docs.bitnami.com/virtual-machine/apps/phabricator/#how-to-upgrade-phabricator) (using `git pull`) and currently it stands as:

> **phabricator** 32a580c720e31b0e4d7c5e8846870e8ce9fe3978 (Sep 8 2016) 
> **arcanist** 10e5194752901959507223c01e0878e6b8312cc5 (Aug 26 2016) 
> **phutil** c14343ee620e43f0ba55eaf80aabcba78c732d45 (Sep 5 2016) 
> **Local Version** Bitnami Phabricator 20160427-0

_(from: Phabricator >  Config > Version Information)_

Now I wish to move this setup to a new box, using the docker images of this project. However, when I tried to restore a backup from my current setup (above versions) in a new `2016.40-r0` docker image I got a plethora of errors (both during the database schema upgrade and during the execution -- most ones about missing columns in db).

So, I believe the simplest way to get this done would be first to upgrade my current setup to the exact version used in `2016.40-r0` docker image and only then make a backup. This way I can restore it in the new box (docker container) without errors.

BTW: I'm following the [Phabricator instructions](https://secure.phabricator.com/book/phabricator/article/configuring_backups/) for the backup/restore cycle. Unfortunately I can`t follow the [Bitnami one](https://docs.bitnami.com/virtual-machine/apps/phabricator/#how-to-create-a-full-backup-of-phabricator) because the paths differ between the Bitnami docker image and Bitnami native installs.

If there is a better path to make this migration, please advise.

**Comment by fdcastel** [comment]
I tried to install a clean `2016.40-r0` image but unfortunately _Phabricator >  Config > Version Information_ shows nothing:

> **phabricator** Unknown 
> **arcanist** Unknown 
> **phutil** Unknown

I also tried to get the version directly from `/opt/bitnami/phabricator` using git commands but this path isn't a git working copy (probably the reason why phabricator can't show the versions)

**Comment by fdcastel** [comment]
I found the problem with the database upgrade after restore. 

I was restoring the mysql dumps OVER the initial (empty) database created by Phabricator. At first I didn't find this could be a problem, however there are some edge cases. E.g.:

```
Applying patch 'phabricator:20160913.conpherence.topic.1.sql'...
Applying patch 'phabricator:20160919.repo.messagecount.sql'...
Applying patch 'phabricator:20160919.repo.messagedefault.sql'...
Applying patch 'phabricator:20160921.fileexternalrequest.sql'...
[2016-10-15 03:13:54] EXCEPTION: (AphrontQueryException) #1050: Table 'file_externalrequest' already exists at [<phutil>/src/aphront/storage/connection/mysql/AphrontBaseMySQLDatabaseConnection.php:355]
arcanist(), phabricator(), phutil()
```

In this case, the table was created by Phabricator 2016.40 initial setup but it didn't exist in the previous version (from which the dump was made). And the db upgrade fails because the table "already exists" in a schema version which it shouldn't.

The solution was to destroy all the databases before the restore with a `bin/storage destroy`.

To sum up: the steps needed to restore a database dump from an older version are:

1. `bin/phd stop`
2. `bin/storage destroy --force`
3. Restore mysql dumps
4. `bin/storage upgrade --force`
5. `bin/phd start`

Just for the sake of clarity: These are NOT the complete steps needed for a full migration. For more detailed explanations, please see: [Configuring Backups and Performing Migrations](https://secure.phabricator.com/book/phabricator/article/configuring_backups/)

P.S.: since this issue is becoming a mess 😄 I'll split it into several others accordingly.

**Comment by fdcastel** [comment]
I'm closing this issue. Please see #14 and #15.

**Comment by chipironcin** [comment]
Thanks for your research @fdcastel and sorry for the inconveniences.
We will be attending your other opened issues.
